### PR TITLE
bulk media upload support for nfs drive

### PR DESCRIPTION
**Pull request by snopoke**
@proteusvacuum 

@sravfeyn buddy ping

### CODE REVIEW

File: corehq/apps/hqmedia/views.py [hunk 1]
```diff
@@ -267,13 +275,18 @@ def validate_file(self):
...
-        status.save()
+        if hasattr(self.uploaded_file, 'temporary_file_path') and settings.SHARED_DRIVE_CONF.temp_dir:
+            processing_id = uuid.uuid4().hex
+            path = '{}.{}'.format(os.path.join(settings.SHARED_DRIVE_CONF.temp_dir, processing_id), 'upload')
+            shutil.move(self.uploaded_file.temporary_file_path(), path)
+            status = BulkMultimediaStatusCacheNfs(processing_id, path)
+            status.save()
+        else:
+            self.uploaded_file.file.seek(0)
+            saved_file = expose_download(self.uploaded_file.file.read(), expiry=BulkMultimediaStatusCache.cache_expiry)
```
Comment (dimagimon) [inline corehq/apps/hqmedia/views.py]: E501 line too long (119 > 115 characters)

File: settings.py [hunk 2]
```diff
@@ -1110,6 +1107,17 @@
 
 MIDDLEWARE_CLASSES += LOCAL_MIDDLEWARE_CLASSES
 
+### Shared drive settings ###
```
Comment (dimagimon) [inline settings.py]: E265 block comment should start with '# '

File: settings.py [hunk 3]
```diff
@@ -1110,6 +1107,17 @@
...
+SHARED_DRIVE_CONF = SharedDriveConfiguration(
+    SHARED_DRIVE_ROOT,
+    RESTORE_PAYLOAD_DIR_NAME,
+    TRANSFER_FILE_DIR_NAME,
+    SHARED_TEMP_DIR_NAME
+)
+TRANSFER_MAPPINGS = {
+    SHARED_DRIVE_CONF.transfer_dir: '/{}'.format(TRANSFER_FILE_DIR_NAME),  # e.g. '/mnt/shared/downloads': '/downloads',
```
Comment (dimagimon) [inline settings.py]: E501 line too long (120 > 115 characters)

File: corehq/apps/hqmedia/cache.py [hunk 4]
```diff
@@ -107,7 +110,43 @@ def add_matched_path(self, media_class, media_info):
...
                                     _("Not a bulk-upload supported CommCareMedia type: %s" % media_class.__name__))
 
+    def _get_upload_file(self):
+        try:
+            saved_file = StringIO.StringIO()
+            saved_ref = DownloadBase.get(self.processing_id)
+            data = saved_ref.get_content()
+            saved_file.write(data)
+            saved_file.seek(0)
+        except Exception as e:
```
Comment (sravfeyn) [inline corehq/apps/hqmedia/cache.py]: Should this be caught one line 116?

Comment (snopoke) [inline corehq/apps/hqmedia/cache.py]: It could be I guess, will update.

**Comment by benrudolph** [comment]
sweeeet

**Comment by czue** [comment]
@sravfeyn did you mean to merge this? looks like it still has a "don't merge" label. cc @snopoke

**Comment by snopoke** [comment]
@sravfeyn No problem to have this merged but in future please do respect the PR labels.

**Comment by sravfeyn** [comment]
hey, sorry. I should have looked at the labels carefully.

**Pull request merged**